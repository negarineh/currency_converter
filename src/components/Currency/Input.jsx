import React from 'react';
import {
    CurrencyConverter,
    Input,
} from "./styles";

import Select from "./Select";

function InputCurrency({
    currencyList,
    inputValue,
    selectValue,
    inputChange,
    selectChange,
    inputRef,
}) {
    return (
        <CurrencyConverter>
            <Input
                type="number"
                value={inputValue}
                onChange={inputChange}
                ref={inputRef}
            />

            <Select
                value={selectValue}
                onChange={selectChange}
                currencyList={currencyList}
            />
        </CurrencyConverter>
    )
}

export default React.memo(InputCurrency);
