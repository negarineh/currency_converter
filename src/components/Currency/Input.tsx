import React from 'react';
import {
    CurrencyConverter,
    Input,
} from "./styles";

import Select from "./Select";
import InputTypes from './Input.types';

function InputCurrency({
    currencyList,
    inputValue,
    selectValue,
    inputChange,
    selectChange,
    inputRef,
}: InputTypes) {
    return (
        <CurrencyConverter>
            <Input
                type="number"
                value={inputValue}
                onChange={inputChange}
                ref={inputRef}
            />

            <Select
                value={selectValue}
                onChange={selectChange}
                currencyList={currencyList}
            />
        </CurrencyConverter>
    )
}

export default React.memo(InputCurrency);
