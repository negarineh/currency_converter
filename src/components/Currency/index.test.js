import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
require('core-js/stable');

configure({ adapter: new Adapter() });

import {
    GlobalStyle,
    AppWrapper,
    CurrencyInfo,
} from "./styles";
import Input from './Input';

import { currencyList } from './Input.test';

const isFetched = true;
const from = 'AUD';
const convertFrom = 0;
const fromChangeInput = jest.fn();
const fromCurrencyChange = jest.fn();
const to = 'AUD';
const convertTo = 0;
const toChangeInput = jest.fn();
const toCurrencyChange = jest.fn();
const value = 0;

test('Render index component', () => {
    const container = renderer.create(
        <>
            <GlobalStyle />
            {isFetched && (
                <AppWrapper>
                    <CurrencyInfo>
                        <p>
                            {value} Australian Dollar equals
                        </p>
                        <h4>
                            {value} United States Dollar
                        </h4>
                    </CurrencyInfo>
                    <Input
                        currencyList={currencyList}
                        inputValue={from}
                        selectValue={convertFrom}
                        inputChange={fromChangeInput}
                        selectChange={fromCurrencyChange}
                    />
                    <Input
                        currencyList={currencyList}
                        inputValue={to}
                        selectValue={convertTo}
                        inputChange={toChangeInput}
                        selectChange={toCurrencyChange}
                    />
                </AppWrapper>
            )}
        </>
    ).toJSON();
    expect(container).toMatchSnapshot();
});

test('it applies styles', () => {
    const appWrapper = shallow(
        <AppWrapper />
    );

    expect(appWrapper).toHaveLength(1);
    expect(appWrapper).toHaveStyleRule('text-align', 'center');
    expect(appWrapper).toHaveStyleRule('background-color', '#fff');
    expect(appWrapper).toHaveStyleRule('border-radius', '8px');
});
