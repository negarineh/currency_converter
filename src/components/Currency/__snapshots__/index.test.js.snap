// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Input /> 1`] = `
.c2 {
  font-size: 0.9rem;
  height: 3rem;
  width: 20rem;
  background-color: #fff;
  border-radius: 5px;
  padding: 0.7rem;
}

.c2:focus {
  outline: none;
}

.c0 {
  margin-bottom: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
}

.c1 {
  width: 14rem;
  padding: 0.7em;
  border: 1px solid #dbe2e8;
  color: #2e3d49;
  max-width: 50rem;
  margin-right: 2.5rem;
  outline: none;
  height: 3rem;
  font-size: 0.9rem;
  border-radius: 5px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
  margin: 1rem;
}

@media only screen and (max-width:600px) {
  .c2 {
    width: 14rem;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    onChange={[MockFunction]}
    type="number"
    value={0}
  />
  <select
    className="c2"
    name="currencyList"
    onChange={[MockFunction]}
    value={0}
  >
    <option
      value="AUD"
    >
      Australian Dollar
    </option>
    <option
      value="CAD"
    >
      Canadian Dollar
    </option>
  </select>
</div>
`;

exports[`Render index component 1`] = `
.c4 {
  font-size: 0.9rem;
  height: 3rem;
  width: 20rem;
  background-color: #fff;
  border-radius: 5px;
  padding: 0.7rem;
}

.c4:focus {
  outline: none;
}

.c0 {
  text-align: center;
  background-color: #fff;
  max-width: 800px;
  margin: 3rem auto 0 auto;
  border-radius: 8px;
  padding: 0.75rem 1rem;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 1.5rem;
  font-family: arial,sans-serif;
}

.c1 > p {
  font-size: 1.2rem;
  margin: 0;
  color: #b0afbb;
  font-family: Roboto,HelveticaNeue,Arial,sans-serif;
}

.c1 > h4 {
  margin: 0 0 0 1.2rem;
  font-size: 1.2rem;
  color: #4b4897;
  font-family: Google Sans,arial,sans-serif;
}

.c2 {
  margin-bottom: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
}

.c3 {
  width: 14rem;
  padding: 0.7em;
  border: 1px solid #dbe2e8;
  color: #2e3d49;
  max-width: 50rem;
  margin-right: 2.5rem;
  outline: none;
  height: 3rem;
  font-size: 0.9rem;
  border-radius: 5px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
  margin: 1rem;
}

@media only screen and (max-width:600px) {
  .c4 {
    width: 14rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      0
       Australian Dollar equals
    </p>
    <h4>
      0
       United States Dollar
    </h4>
  </div>
  <div
    className="c2"
  >
    <input
      className="c3"
      onChange={[MockFunction]}
      type="number"
      value="AUD"
    />
    <select
      className="c4"
      name="currencyList"
      onChange={[MockFunction]}
      value={0}
    >
      <option
        value="AUD"
      >
        Australian Dollar
      </option>
      <option
        value="CAD"
      >
        Canadian Dollar
      </option>
    </select>
  </div>
  <div
    className="c2"
  >
    <input
      className="c3"
      onChange={[MockFunction]}
      type="number"
      value="AUD"
    />
    <select
      className="c4"
      name="currencyList"
      onChange={[MockFunction]}
      value={0}
    >
      <option
        value="AUD"
      >
        Australian Dollar
      </option>
      <option
        value="CAD"
      >
        Canadian Dollar
      </option>
    </select>
  </div>
</div>
`;
